extensions/notebook/src/jupyter/jupyterServerManager.ts

```diff
@@ -118,17 +118,36 @@ export class LocalJupyterServerManager implements nb.ServerManager, vscode.Dispo
 		// /path2/nb2.ipynb
 		// /path2/nb3.ipynb
 		// ... will result in 2 notebook servers being started, one for /path1/ and one for /path2/
-		let notebookDir = this.apiWrapper.getWorkspacePathFromUri(vscode.Uri.file(this.documentPath));
-		if (!notebookDir) {
-			let docDir = path.dirname(this.documentPath);
-			if (docDir === '.') {
-				// If the user is using a system version of python, then
-				// '.' will try to create a notebook in a system directory.
-				// Since this will fail due to permissions, use the user's
-				// home folder instead.
-				notebookDir = utils.getUserHome();
+		let notebookDir = this._apiWrapper.getWorkspacePathFromUri(vscode.Uri.file(this.documentPath));
+		let docDir;
+		if (notebookDir) {
+			docDir = path.dirname(notebookDir);
+		} else {
+			docDir = path.dirname(this.documentPath);
+		}
+		let parsedPath = path.parse(docDir);
+		let userHome = utils.getUserHome();
+		// If you don't match the docDir's casing for drive letters, a 404 will result
```

Comment:
it seems this block of code is Windows specific, to be safe, I'd recommend to add a OS check here.

Reply:
for linux and mac only one instance of Jupyter is required, right?

Reply:
It's still the same behavior unfortunately